packages/rewards/ISSUES.md

```diff
@@ -0,0 +1,79 @@
+# Analysis
+
+This document outlines all potential issues with the `@recallnet/rewards` package implementation.
+
+## Issues
+
+### 1. **Mathematical Formula Edge Cases**
+
+- **Issue**: The decay formula `w_i = ((1-r) * r^(i-1)) / (1-r^k)` can have numerical instability
+- **Risk**: When `r` approaches 1 or 0, or when `k` is very large, the formula can produce unexpected results
+- **Impact**: Could break reward distribution entirely
+- **Mitigation**: Add bounds checking and fallback calculations
+
+### 2. **Tie Handling**
+
+- **Issue**: No logic for handling tied competitors
+- **Risk**: Undefined behavior when competitors have equal scores
+- **Impact**: Unpredictable results
+- **Mitigation**: Define tie-breaking rules
+
+### 3. **Reward Distribution Completeness**
+
+- **Issue**: The spec mentions that total payouts may not equal the prize pool due to time decay
+- **Risk**: Users might not receive the full prize pool
+- **Impact**: User dissatisfaction
+- **Mitigation**: Document this behavior clearly and consider redistribution strategies
+
+### 4. **No Audit Trail**
+
+- **Issue**: No logging or verification of calculations
+- **Risk**: Difficult to debug or verify results
+- **Impact**: Trust and debugging issues
+- **Mitigation**: Add comprehensive logging
+
+### 5. **Deterministic but Unverifiable**
+
+- **Issue**: While calculations are deterministic, there's no way to verify them externally
+- **Risk**: Users can't independently verify their rewards
+- **Impact**: Trust issues
+- **Mitigation**: Provide verification tools or APIs
+
+## Performance and Scalability Issues
+
+### 6. **O(n²) Algorithm Complexity**
+
+- **Issue**: The main calculation loops through all users for each competitor
+- **Risk**: Performance degrades quadratically with number of users/competitors
+- **Impact**: Could become slow with large competitions
+- **Mitigation**: Optimize algorithm or add performance monitoring
+
+### 7. **Memory Usage**
```

Comment:
I have measured the memory usage to calculate rewards for 300k users, it takes roughly 800MB. It's quite a lot, but we can handle it. But it is definitely something to be aware of.